fw_gear_hcp_diff/diff_main.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+import os
+import os.path as op
+
+import flywheel
+
+from fw_gear_hcp_diff.args import DiffPreprocPipeline, hcpdiff_qc_mosaic
+from utils import diff_utils, gear_preliminaries, results
+
+
+def run(context):
+    """
+    Set up and complete the Diffusion Processing stage in the HCP Pipeline.
+    """
+    # Preamble: take care of all gear-typical activities.
+    context.gear_dict = {}
+    # Initialize all hcp-gear variables.
+    gear_preliminaries.initialize_gear(context)
+
+    context.log_config()
+
+    # Utilize FreeSurfer license from config or project metadata.
+    try:
+        gear_preliminaries.set_freesurfer_license(context)
+    except Exception as e:
+        context.log.exception(e)
+        context.log.fatal(
+            "A valid FreeSurfer license must be present to run. "
+            "Please check your configuration and try again."
+        )
+        os.sys.exit(1)
+
+    # Before continuing from here, we need to validate the config.json
+    # Validate gear configuration against gear manifest
+    try:
+        gear_preliminaries.validate_config_against_manifest(context)
+    except Exception as e:
+        context.log.error("Invalid Configuration:")
+        context.log.exception(e)
+        context.log.fatal("Please make the prescribed corrections and try again.")
+        os.sys.exit(1)
+
+    # Get file list and configuration from hcp-struct zipfile
+    try:
+        hcp_struct_zip_filename = context.get_input_path("StructZip")
+        hcp_struct_list, hcp_struct_config = gear_preliminaries.preprocess_hcp_zip(
+            hcp_struct_zip_filename
+        )
+        context.gear_dict["exclude_from_output"] = hcp_struct_list
+        context.gear_dict["hcp_struct_config"] = hcp_struct_config
+    except Exception as e:
+        context.log.exception(e)
+        context.log.error("Invalid hcp-struct zip file.")
+        os.sys.exit(1)
+
+    # Ensure the subject_id is set in a valid manner
+    # (api, config, or hcp-struct config)
+    try:
+        gear_preliminaries.set_subject(context)
+    except Exception as e:
+        context.log.exception(e)
+        context.log.fatal(
+            "The Subject ID is not valid. Examine and try again.",
+        )
+        os.sys.exit(1)
+
+    ############################################################################
+    # Build and Validate Parameters
+    # Doing as much parameter checking before ANY computation.
+    # Fail as fast as possible.
+
+    try:
+        # Build and validate from Volume Processing Pipeline
+        DiffPreprocPipeline.build(context)
```

Comment:
I would avoid setting values in the context itself and instead have this function return a list of params.